src/app-host/app-host.js

```diff
@@ -5,93 +5,184 @@ var Messages = require('messages');
+        // the list of plugins
+        socket.on('exec-success', function (data) {
+            var execCacheInfo = execCache[data.index];
+            if (execCacheInfo && typeof execCacheInfo.success === 'function') {
+                execCacheInfo.success(data.result);
+            }
+        });
 
-socket.on('exec-success', function (data) {
-    console.log('exec-success:');
-    console.log(data);
-    var execCacheInfo = execCache[data.index];
-    if (execCacheInfo.success) {
-        execCacheInfo.success(data.result);
-    }
-});
+        socket.on('exec-failure', function (data) {
+            var execCacheInfo = execCache[data.index];
+            if (execCacheInfo && typeof execCacheInfo.fail === 'function') {
+                execCacheInfo.fail(data.error);
+            }
+        });
 
-socket.on('exec-failure', function (data) {
-    console.log('exec-failure:');
-    console.log(data);
-    var execCacheInfo = execCache[data.index];
-    if (execCacheInfo.fail) {
-        execCacheInfo.fail(data.error);
-    }
-});
+        socket.on('start-live-reload', function () {
+            livereload.start(socket);
+        });
 
-socket.on('start-live-reload', function () {
-    livereload.start(socket);
-});
+        socket.on('init-telemetry', function (data) {
+            telemetry.init(socket);
+        });
 
-socket.on('init-telemetry', function (data) {
-    telemetry.init(socket);
-});
+        socket.on('init-xhr-proxy', function (data) {
+            require('xhr-proxy').init(); 
+        });
 
-socket.on('init-xhr-proxy', function (data) {
-    require('xhr-proxy').init();
-});
+        socket.on('init-touch-events', function (data) {
+            require('./touch-events').init();
+        });
 
-socket.on('init-touch-events', function (data) {
-    require('./touch-events').init();
-});
+        if (cordova.platformId !== 'browser') {
+            channel.onPluginsReady.subscribe(function () {
+                var pluginList = cordova.require('cordova/plugin_list').metadata;
+                socket.emit('app-plugin-list', pluginList);
+            });
+        } else {
+            socket.emit('app-plugin-list', {});
+        }
+
+        applyPlugins(plugins);
+        applyPlugins(pluginHandlersDefinitions, pluginHandlers, serviceToPluginMap);
+        applyPlugins(pluginClobberDefinitions, window);
+
+        platformBootstrap();
+
+        switch (cordova.platformId) {
+            // these platform fire onNativeReady in their bootstrap
+            case 'ios':
+            case 'browser':
+            case 'blackberry10':
+            case 'firefoxos':
+            case 'ubuntu':
+            case 'webos':
+                break;
+            // windows has an overriden bootstrap which does not fire
+            // onNativeReady
+            case 'windows':
+            /* falls through */
+            default:
+                channel.onNativeReady.fire();
```

Comment:
Since `windows` is already there with a `/* falls through */` comment, I think it's a good idea to add `android` as well